File: app/src/main/java/com/uolinc/marvelapp/network/ApiCall.java
package com.uolinc.marvelapp.network;

import com.uolinc.marvelapp.model.ReturnData;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Interface responsável para fazer a chamada com a API
 */
public interface ApiCall {

    @GET("characters")
    Call<ReturnData> listCharacter(@Query("ts") String timeStamp,
                                   @Query("apikey") String apiKey,
                                   @Query("hash") String hash,
                                   @Query("limit") int limit,
                                   @Query("offset") int offset,
                                   @Query("orderBy") String orderBy);

}

File: app/src/main/java/com/uolinc/marvelapp/network/ApiCall.kt
package com.uolinc.marvelapp.network

import com.uolinc.marvelapp.model.response.DataResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Interface responsável para fazer a chamada com a API
 */
interface ApiCall {

    @GET("characters")
    fun listCharacter(@Query("ts") timeStamp: String,
                      @Query("apikey") apiKey: String,
                      @Query("hash") hash: String,
                      @Query("offset") offset: Int = 0,
                      @Query("orderBy") orderBy: String): Observable<DataResponse>
}